google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java

```diff
@@ -552,10 +556,12 @@ public void testDeleteRecursive() throws IOException {
 
   @Test
   public void testListFilesInRootDirectory() throws IOException {
+    // We must explicitly set the storageOptions, because the unit tests
+    // set the fake storage as default but we want to access the real storage.
```

Comment:
If this needs to test against real storage, shouldn't it go into `ITStorageTest`?

Reply:
This test case accesses real storage using the NIO interface, so it is correct for it to be with the others in `ITGcsNio` and not `ITStorageTest`.

Reply:
Sorry I got confused. Since this is `ItGcsNio`, shouldn't other tests be using the real storage server anyway?

Pardon my ignorance, I'm still trying to page things back from my long term memory.

Reply:
The other tests do too. They all specify `storageOptions` - this test was the only outlier, a situation that this pull request aims to resolve.

The unit tests mentioned in the comment are outside of ITGcsNio. But because they work by changing the default (something that's normally done only once per program), when running the full test suite the integration test must not rely on the default.

A different way to fix this would be to change all the unit tests to explicitly set storageOptions - however the unit tests check the various ways to create an NIO cloud provider (not all of which allow explicit options). A further alternative would be to reset the defaults when running the integration tests. However since the integration test really should be using storageOptions to set lenient retry settings (we want it to retry if GCS is transiently slow rather than erroneously reporting a failed test), I favor the solution in this PR: set storageOptions for all of the NIO integration tests.

Reply:
You're right. I stand corrected.